lib/xcodebuild.js

```diff
@@ -249,22 +254,26 @@ export class XcodeBuild {
     if (this.useXctestrunFile && this.xctestrunFilePath) {
       args.push('-xctestrun', this.xctestrunFilePath);
     } else {
-      const runnerScheme = isTvOS(this.platformName) ? RUNNER_SCHEME_TV : RUNNER_SCHEME_IOS;
+      const runnerScheme = isTvOS(this.platformName || '') ? RUNNER_SCHEME_TV : RUNNER_SCHEME_IOS;
       args.push('-project', this.agentPath, '-scheme', runnerScheme);
       if (this.derivedDataPath) {
         args.push('-derivedDataPath', this.derivedDataPath);
       }
     }
     args.push('-destination', `id=${this.device.udid}`);
 
-    const versionMatch = new RegExp(/^(\d+)\.(\d+)/).exec(this.platformVersion);
-    if (versionMatch) {
-      args.push(
-        `${isTvOS(this.platformName) ? 'TV' : 'IPHONE'}OS_DEPLOYMENT_TARGET=${versionMatch[1]}.${versionMatch[2]}`
-      );
+    if (this.platformVersion) {
+      const versionMatch = new RegExp(/^(\d+)\.(\d+)/).exec(this.platformVersion);
+      if (versionMatch) {
+        args.push(
+          `${isTvOS(this.platformName || '') ? 'TV' : 'IPHONE'}OS_DEPLOYMENT_TARGET=${versionMatch[1]}.${versionMatch[2]}`
+        );
+      }
     } else {
-      this.log.warn(`Cannot parse major and minor version numbers from platformVersion "${this.platformVersion}". ` +
-        'Will build for the default platform instead');
+      this.log.warn(
```

Comment:
Maybe the code wants to print the log for `else` for `if (versionMatch) {` case as well..?